llvm/lib/IR/RepoTicket.cpp

```diff
@@ -77,76 +78,55 @@ const Constant *getAliasee(const GlobalAlias *GA) {
   return Target;
 }
 
-static const DependenciesType &
-updateInitialDigestAndGetDependencies(const GlobalObject *GO, MD5 &GOHash,
-                                      GOInfoMap &GOIMap, GONumber &GONum) {
-  GOInfoMap::const_iterator Pos;
+static const GODigestState
+updateInitialDigestAndGetGOStatus(const GlobalObject *GO, MD5 &GOHash,
```

Comment:
It sounds like this function is doing more than one thing (and was before to be fair)... I think it's generally recommended for a function to "do one thing" so this could be refactored (or possibly it's really doing one thing and a different name could express that).

Reply:
And why is this named "...AndGetGOStatus" where the other two related functions are named "...AndGetGOState"?

Reply:
> 
> 
> It sounds like this function is doing more than one thing (and was before to be fair)... I think it's generally recommended for a function to "do one thing" so this could be refactored (or possibly it's really doing one thing and a different name could express that).

This function has been split into two functions named `calculateGOInfo` and `accumulateGOInitialDigestAndGetGODigestState`.

> 
> 
> And why is this named "...AndGetGOStatus" where the other two related functions are named "...AndGetGOState"?

Changed to `...GetGOState`.